src/main/java/com/fundynamic/d2tm/game/controls/Mouse.java

```diff
@@ -61,13 +72,36 @@ public void deselectEntity() {
                 ((Selectable) lastSelectedEntity).deselect();
             }
         }
+        if (selectedEntityBelongsToPlayerControlled()) {
+            movingCursor = false;
+        }
         lastSelectedEntity = null;
     }
 
+    private boolean selectedEntityBelongsToPlayerControlled() {
+        if (lastSelectedEntity == null) return false;
+        if (playerControlled == null) return false;
```

Comment:
Yes mouse states makes sense. Switching between them would be when clicking on a unit or something. Lets do this before merging.